modules/auxiliary/gather/behind_ingensec.rb

```diff
@@ -0,0 +1,286 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Behind BinarySec/IngenSecurity',
+      'Version' => '$Release: 1.0',
+      'Description'    => %q{
+        This module can be useful if you need to test
+        the security of your server and your website
+        behind BinarySec/IngenSec by discovering the real IP address.
+      },
+      'Author'         => [
+        'mekhalleh',
+        'RAMELLA Sébastien <sebastien.ramella[at]Pirates.RE>'
+      ],
+      'References' => [
+        ['URL', 'https://www.ingensecurity.com/fct_apercu.html']
+      ],
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        OptString.new('HOSTNAME', [true, 'The hostname to find real IP', 'www.reunionnaisdumonde.com']),
+        OptString.new('URIPATH', [true, 'The URI path (for custom comparison)', '/']),
+        OptInt.new('RPORT', [true, 'The target port (for custom comparison)', 443]),
+        OptBool.new('SSL', [true, 'Negotiate SSL/TLS for outgoing connections (for custom comparison)', true]),
+        OptString.new('Proxies', [false, 'A proxy chain of format type:host:port[,type:host:port][...]']),
+        OptInt.new('THREADS', [false, 'Threads for DNS enumeration', 1]),
+        OptPath.new('WORDLIST', [true, 'Wordlist of subdomains', ::File.join(Msf::Config.data_directory, 'wordlists', 'namelist.txt')])
+      ])
+  end
+
+  def do_check_tcp_port(ip, port, proxies)
+    begin
+      sock = Rex::Socket::Tcp.create(
+        'PeerHost' => ip,
+        'PeerPort' => port,
+        'Proxies'  => proxies
+      )
+    rescue ::Rex::ConnectionRefused, Rex::ConnectionError
+      return(false)
+    end
+    sock.close
+    return(true)
+  end
+
+  def do_grab_domain_ip_history(hostname, proxies)
+    begin
+      cli = Rex::Proto::Http::Client.new('www.prepostseo.com', 443, {}, true, nil, proxies)
```

Comment:
Consider using `send_request_cgi` ?

Reply:
thanks for the review.
I'm not sure I understand that point. Can you tell me more about this?

Reply:
`send_request_cgi` and `send_request_raw` are the good wrapper for CGI and arbitrary HTTP requests, which could make the code cleaner and more readable.

See doc [here](https://rapid7.github.io/metasploit-framework/api/Msf/Exploit/Remote/HttpClient.html#send_request_cgi-instance_method)

Reply:
I keep this reference for later (I have to implement other features in the future).

I made a code update and a commit.

PS: add. new auxiliary module: behind_cloudflare # 10481 is also based on bypass, if you can also do a review? I anticipated the synthaxic corrections.

Thanks.